### DISCUSSION TITLE
failure in writing quacc data to database using slurm job

### DISCUSSION THREAD

**Discussion by yw-fang** [Quantum-Accelerators/quacc#2508] [General]
This issue was relevant to https://github.com/Quantum-Accelerators/quacc/discussions/2502#discussioncomment-10993639

I am trying to let quacc be interacted with maongodb, but unsure if the following issue was caused by quacc or mongodb itself or just my incorrect workflow. If it's not appropriate to propose this issue here, please feel free to let me know about it.

The example code by Andrew in https://github.com/Quantum-Accelerators/quacc/discussions/2502#discussioncomment-10993639 could carry out the VASP calculations by submitting it via slurm, but it raised errors when connecting to the database:
```
  File "/packages/pymongo/synchronous/topology.py", line 333, in _select_servers_loop
    raise ServerSelectionTimeoutError(
pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 30s, Topology Description: <TopologyDescription id: 671449cab699f8c15e9ecb57, topology_type: Unknown, servers: [<ServerDescription ('localhost', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('localhost:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
```

However, if I wrote a simple python script "try.py" and run it directly using "python try.py", it could write to the database.

try.py:
```
from pymongo import MongoClient, errors
try:
    client = MongoClient('mongodb://localhost:27017/', serverSelectionTimeoutMS=5000)  # 5 seconds timeout
    db = client['mydb']
    print("Successfully connected to MongoDB!")

    collection = db['outputs']

    test_doc = {"message": "Hello from Yuewen!"}
    insert_result = collection.insert_one(test_doc)

    print("Inserted document ID:", insert_result.inserted_id)

except errors.ServerSelectionTimeoutError as err:
    print("Failed to connect to MongoDB:", err)
except Exception as e:
    print("An error occurred:", e)
```

This simple script returns "Successfully connected to MongoDB!
Inserted document ID: 671446cc4c5ce16433exxxxxx
"
I crosschecked by mongosh and found it was indeed in the database:
```
[
  {
    _id: ObjectId('671446cc4c5ce16433exxxxxx'),
    message: 'Hello from Yuewen!'
  }
]

```
Could an expert make any comments? Thank you very much!

**Comment by yw-fang** [comment]
Could it be that the SLURM job is running on a compute node that cannot connect to the database? However, when I run the script directly using 'python try.py', it is on the login node that can access the database (note that the database is on my local computer, and the remote HPC uses an SSH tunnel to connect to it).

**Reply by yw-fang** [reply]
I confirmed my assumption was correct. By setting up the tunnel between the compute node and the database within the SLURM job script, it successfully allowed quacc to write data to the database. Thanks so much for your attention, and I hope this will be useful for other future users as well.

**Reply by Andrew-S-Rosen** [reply]
Yup, that is indeed the case. Thanks for sharing your thought process here.
